File: gin/modules/admin/web_news.go
package admin

import (
	"webmis/config"
	"webmis/library"
	"webmis/model"
	"webmis/service"
	"webmis/util"

	"github.com/gin-gonic/gin"
)

var ImgDir string = "upload/news/"

type WebNews struct {
	service.Base
}

/* 列表 */
func (r WebNews) List(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	data, _ := r.JsonName(json, "data")
	page, _ := r.JsonName(json, "page")
	limit, _ := r.JsonName(json, "limit")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(data) || util.Empty(page) || util.Empty(limit) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 条件
	param := map[string]interface{}{}
	util.JsonDecode(data, &param)
	title := util.Trim(util.If(util.InKey("title", param), param["title"], ""))
	// 统计
	m := (&model.WebNews{}).New()
	m.Columns("count(*) AS num")
	m.Where("title LIKE ?", "%"+title+"%")
	total := m.FindFirst()
	// 查询
	m.Columns("id", "cid", "title", "source", "author", "FROM_UNIXTIME(ctime, '%Y-%m-%d %H:%i:%s') as ctime", "FROM_UNIXTIME(utime, '%Y-%m-%d %H:%i:%s') as utime", "state", "img", "summary")
	m.Where("title LIKE ?", "%"+title+"%")
	m.Page((&util.Type{}).Int(page), (&util.Type{}).Int(limit))
	m.Order("id DESC")
	list := m.Find()
	// 数据
	for _, v := range list {
		v["img"] = (&service.Data{}).Img(v["img"])
		if (&util.Type{}).Strval(v["state"]) == "1" {
			v["state"] = true
		} else {
			v["state"] = false
		}
	}
	// 返回
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功", "list": list, "total": (&util.Type{}).Int(total["num"])})
}

/* 添加 */
func (r WebNews) Add(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	data, _ := r.JsonName(json, "data")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(data) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 数据
	param := map[string]interface{}{}
	util.JsonDecode(data, &param)
	base64 := util.Trim(util.If(util.InKey("img", param), param["img"], ""))
	cid := util.Trim(util.If(util.InKey("cid", param), param["cid"], ""))
	title := util.Trim(util.If(util.InKey("title", param), param["title"], ""))
	source := util.Trim(util.If(util.InKey("source", param), param["source"], ""))
	author := util.Trim(util.If(util.InKey("author", param), param["author"], ""))
	summary := util.Trim(util.If(util.InKey("summary", param), param["summary"], ""))
	if base64 == "" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "请上传封面图!"})
		return
	}
	if cid == "" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "请选择分类!"})
		return
	}
	if util.Len(title) < 2 || util.Len(title) > 30 {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "新闻标题2～30字符!"})
		return
	}
	// 封面图
	path := ImgDir + "img/"
	img := (&library.Upload{}).Base64(map[string]interface{}{"path": path, "base64": base64})
	// 模型
	db := (&model.WebNews{}).New()
	conn := db.DBConn()
	tx, _ := conn.Begin()
	// 信息
	m1 := (&model.WebNews{}).New()
	m1.Values(map[string]interface{}{"cid": cid, "title": title, "source": source, "author": author, "summary": summary, "ctime": util.Time(), "utime": util.Time(), "img": path + img})
	sql, args := m1.InsertSQL()
	rs, err1 := tx.Exec(sql, args...)
	id := db.LastInsertId(rs)
	// 内容
	m2 := (&model.WebNewsHtml{}).New()
	m2.Values(map[string]interface{}{"nid": id})
	sql, args = m2.InsertSQL()
	_, err2 := tx.Exec(sql, args...)
	if err1 != nil || err2 != nil {
		tx.Rollback()
		(&library.FileEo{}).New(config.Env().RootDir)
		(&library.FileEo{}).RemoveAll(path + img)
		r.GetJSON(c, gin.H{"code": 5000, "msg": "添加失败!"})
	} else {
		tx.Commit()
		r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
	}
}

/* 编辑 */
func (r WebNews) Edit(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	id, _ := r.JsonName(json, "id")
	data, _ := r.JsonName(json, "data")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(id) || util.Empty(data) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 数据
	param := map[string]interface{}{}
	util.JsonDecode(data, &param)
	base64 := util.Trim(util.If(util.InKey("img", param), param["img"], ""))
	cid := util.Trim(util.If(util.InKey("cid", param), param["cid"], ""))
	title := util.Trim(util.If(util.InKey("title", param), param["title"], ""))
	source := util.Trim(util.If(util.InKey("source", param), param["source"], ""))
	author := util.Trim(util.If(util.InKey("author", param), param["author"], ""))
	summary := util.Trim(util.If(util.InKey("summary", param), param["summary"], ""))
	if base64 == "" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "请上传封面图!"})
		return
	}
	if cid == "" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "请选择分类!"})
		return
	}
	if util.Len(title) < 2 || util.Len(title) > 30 {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "新闻标题2～30字符!"})
		return
	}
	// 封面图
	img := ""
	path := ImgDir + "img/"
	(&library.FileEo{}).New(config.Env().RootDir)
	if base64[0:4] != "http" {
		img = (&library.Upload{}).Base64(map[string]interface{}{"path": path, "base64": base64})
		// 清理封面
		m1 := (&model.WebNews{}).New()
		m1.Columns("img")
		m1.Where("id=?", id)
		tmp := m1.FindFirst()
		(&library.FileEo{}).RemoveAll(tmp["img"].(string))
	}
	// 模型
	m := (&model.WebNews{}).New()
	sData := map[string]interface{}{"cid": cid, "title": title, "source": source, "author": author, "summary": summary, "utime": util.Time()}
	if img != "" {
		sData["img"] = path + img
	}
	m.Set(sData)
	m.Where("id=?", id)
	if m.Update() {
		r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
	} else {
		if img != "" {
			(&library.FileEo{}).RemoveAll(path + img)
		}
		r.GetJSON(c, gin.H{"code": 5000, "msg": "更新失败!"})
	}
}

/* 删除 */
func (r WebNews) Del(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	data, _ := r.JsonName(json, "data")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(data) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 数据
	param := []string{}
	util.JsonDecode(data, &param)
	ids := util.Implode(",", param)
	// 封面图
	m := (&model.WebNews{}).New()
	m.Columns("id", "img")
	m.Where("id in(" + ids + ")")
	imgList := m.Find()
	// 模型
	db := (&model.WebNews{}).New()
	conn := db.DBConn()
	tx, _ := conn.Begin()
	// 信息
	m1 := (&model.WebNews{}).New()
	m1.Where("id in(" + ids + ")")
	sql, args := m1.DeleteSQL()
	_, err1 := tx.Exec(sql, args...)
	// 内容
	m2 := (&model.WebNewsHtml{}).New()
	m2.Where("nid in(" + ids + ")")
	sql, args = m2.DeleteSQL()
	_, err2 := tx.Exec(sql, args...)
	if err1 != nil || err2 != nil {
		tx.Rollback()
		r.GetJSON(c, gin.H{"code": 5000, "msg": "添加失败!"})
	} else {
		tx.Commit()
		// 清理图片
		(&library.FileEo{}).New(config.Env().RootDir)
		for _, v := range imgList {
			(&library.FileEo{}).RemoveAll(v["img"].(string))
			(&library.FileEo{}).RemoveAll(ImgDir + v["id"].(string) + "/")
		}
		r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
	}
}

/* 权限 */
func (r WebNews) State(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	id, _ := r.JsonName(json, "id")
	state, _ := r.JsonName(json, "state")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(id) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 模型
	m := (&model.WebNews{}).New()
	m.Set(map[string]interface{}{"state": util.If((&util.Type{}).Bool(state), "1", "0"), "utime": util.Time()})
	m.Where("id=?", id)
	if m.Update() {
		r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
	} else {
		r.GetJSON(c, gin.H{"code": 5000, "msg": "更新失败!"})
	}
}

/* 分类-获取 */
func (r WebNews) GetClass(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, "")
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	// 查询
	m := (&model.WebNewsClass{}).New()
	m.Columns("id", "name")
	m.Where("state=?", "1")
	m.Order("sort DESC")
	list := m.Find()
	// 数据
	data := []map[string]interface{}{}
	for _, v := range list {
		data = append(data, map[string]interface{}{"label": v["name"], "value": v["id"]})
	}
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功", "data": data})
}

/* 内容-获取 */
func (r WebNews) GetContent(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	nid, _ := r.JsonName(json, "id")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, "")
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	// 查询
	m := (&model.WebNewsHtml{}).New()
	m.Columns("content")
	m.Where("nid=?", nid)
	data := m.FindFirst()
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功", "content": data["content"]})
}

/* 内容-修改 */
func (r WebNews) Content(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	data, _ := r.JsonName(json, "data")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, "")
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(data) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 数据
	param := map[string]interface{}{}
	util.JsonDecode(data, &param)
	id := util.Trim(util.If(util.InKey("id", param), param["id"], ""))
	content := util.Trim(util.If(util.InKey("content", param), param["content"], ""))
	// 图片回收
	(&library.Upload{}).HtmlImgClear(content, ImgDir+id+"/")
	// 模型
	m := (&model.WebNewsHtml{}).New()
	m.Set(map[string]interface{}{"content": content})
	m.Where("nid=?", id)
	if m.Update() {
		r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
	} else {
		r.GetJSON(c, gin.H{"code": 5000, "msg": "更新失败!"})
	}
}

/* 内容-图片 */
func (r WebNews) UpImg(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	base64, _ := r.JsonName(json, "base64")
	id, _ := r.JsonName(json, "id")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, "")
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if util.Empty(base64) {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 上传
	path := ImgDir + id + "/"
	img := (&library.Upload{}).Base64(map[string]interface{}{"path": path, "base64": base64})
	if img != "" {
		r.GetJSON(c, gin.H{"code": 0, "msg": "成功", "img": (&service.Data{}).Img(path + img)})
	} else {
		r.GetJSON(c, gin.H{"code": 5000, "msg": "更新失败!"})
	}
}

File: phalcon/modules/admin/WebNews.php
<?php
namespace App\Admin;

use Service\Base;
use Service\AdminToken;
use Service\Data;
use Model\WebNews as WebNewsM;
use Model\WebNewsHtml;
use Model\WebNewsClass;
use Library\FileEo;
use Library\Upload;
use Util\Util;

class WebNews extends Base {

  private static $ImgDir = 'upload/news/';

  /* 列表 */
	static function List() {
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $data = self::JsonName($json, 'data');
    $page = self::JsonName($json, 'page');
    $limit = self::JsonName($json, 'limit');
    $order = self::JsonName($json, 'order');
    // 验证
    $msg = AdminToken::Verify($token, $_SERVER['REQUEST_URI']);
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($data) || empty($page) || empty($limit)) {
      return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    }
    // 条件
    $param = json_decode($data);
    list($where, $whereData) = self::getWhere($param);
    // 统计
    $m = new WebNewsM();
    $m->Columns('count(*) AS num');
    $m->Where($where, ...$whereData);
    $total = $m->FindFirst();
    // 查询
    $m->Columns('id', 'cid', 'title', 'source', 'author', 'FROM_UNIXTIME(ctime) as ctime', 'FROM_UNIXTIME(utime) as utime', 'state', 'img', 'summary');
    $m->Where($where, ...$whereData);
    $m->Page($page, $limit);
    $m->Order($order?:'id DESC');
    $list = $m->Find();
    self::Print($m->GetSql());
    // 数据
    foreach ($list as $k => $v) {
      $list[$k]['img'] = Data::Img($v['img']);
      $list[$k]['state'] = $v['state']=='1'?true:false;
    }
    // 返回
    return self::GetJSON(['code'=>0,'msg'=>'成功','list'=>$list,'total'=>(int)$total['num']]);
  }
  /* 搜索条件 */
  static private function getWhere(object $param): array {
    // 参数
    $cid = isset($param->cid)?trim($param->cid):'';
    $title = isset($param->title)?trim($param->title):'';
    $source = isset($param->source)?trim($param->source):'';
    $author = isset($param->author)?trim($param->author):'';
    // 条件
    $where = 'cid like ? AND title like ? AND source like ? AND author like ?';
    $whereData = ['%'.$cid.'%', '%'.$title.'%', '%'.$source.'%', '%'.$author.'%'];
    return [$where, $whereData];
  }

  /* 添加 */
  static function Add() {
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $data = self::JsonName($json, 'data');
    // 权限
    $msg = AdminToken::Verify($token, $_SERVER['REQUEST_URI']);
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($data)) {
      return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    }
    // 数据
    $param = json_decode($data);
    $base64 = isset($param->img)?trim($param->img):'';
    $cid = isset($param->cid)?trim($param->cid):'';
    $title = isset($param->title)?trim($param->title):'';
    $source = isset($param->source)?trim($param->source):'';
    $author = isset($param->author)?trim($param->author):'';
    $summary = isset($param->summary)?trim($param->summary):'';
    // 验证
    if($base64=='') return self::GetJSON(['code'=>4000, 'msg'=>'请上传封面图!']);
    if($cid=='') return self::GetJSON(['code'=>4000, 'msg'=>'请选择分类!']);
    if(Util::Len($title)<2 || Util::Len($title)>30) return self::GetJSON(['code'=>4000, 'msg'=>'新闻标题2～30字符!']);
    // 封面图
    $path = self::$ImgDir.'img/';
    $img = Upload::Base64(['path'=>$path, 'base64'=>$base64]);
    // 模型
    $model = new WebNewsM();
    $conn = $model->DBConn();
    try {
      $conn->beginTransaction();
      // 信息
      $m1 = new WebNewsM();
      $m1->Values(['cid'=>$cid, 'title'=>$title, 'source'=>$source, 'author'=>$author, 'summary'=>$summary, 'ctime'=>time(), 'utime'=>time(), 'img'=>$path.$img]);
      list($sql, $args) = $m1->InsertSQL();
      $model->Exec($conn, $sql, $args);
      $id = $model->LastInsertId($conn);
      // 内容
      $m2 = new WebNewsHtml();
      $m2->Values(['nid'=>$id]);
      list($sql, $args) = $m2->InsertSQL();
      $model->Exec($conn, $sql, $args);
      // 提交
      $conn->commit();
      $res = ['code'=>0,'msg'=>'成功'];
    } catch (\Exception $e) {
      $conn->rollBack();
      FileEo::RemoveAll($path.$img);
      $res = ['code'=>5000,'msg'=>'添加失败!'];
    }
    return self::GetJSON($res);
  }

  /* 编辑 */
  static function Edit() {
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $id = self::JsonName($json, 'id');
    $data = self::JsonName($json, 'data');
    // 验证
    $msg = AdminToken::Verify($token, $_SERVER['REQUEST_URI']);
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($id) || empty($data)) {
      return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    }
    // 数据
    $param = json_decode($data);
    $base64 = isset($param->img)?trim($param->img):'';
    $cid = isset($param->cid)?trim($param->cid):'';
    $title = isset($param->title)?trim($param->title):'';
    $source = isset($param->source)?trim($param->source):'';
    $author = isset($param->author)?trim($param->author):'';
    $summary = isset($param->summary)?trim($param->summary):'';
    // 验证
    if($base64=='') return self::GetJSON(['code'=>4000, 'msg'=>'请上传封面图!']);
    if($cid=='') return self::GetJSON(['code'=>4000, 'msg'=>'请选择分类!']);
    if(Util::Len($title)<2 || Util::Len($title)>30) return self::GetJSON(['code'=>4000, 'msg'=>'新闻标题2～30字符!']);
    // 封面图
    $img = '';
    if(substr($base64,0,4)!='http'){
      $path = self::$ImgDir.'img/';
      $img = Upload::Base64(['path'=>$path, 'base64'=>$base64]);
      // 清理封面
      $m1 = new WebNewsM();
      $m1->Columns('img');
      $m1->Where('id=?', $id);
      $tmp = $m1->FindFirst();
      FileEo::RemoveAll($tmp['img']);
    }
    // 模型
    $m = new WebNewsM();
    $data = ['cid'=>$cid, 'title'=>$title, 'source'=>$source, 'author'=>$author, 'summary'=>$summary, 'utime'=>time()];
    if($img!='') $data['img'] = $path.$img;
    $m->Set($data);
    $m->Where('id=?', $id);
    if($m->Update()) {
      return self::GetJSON(['code'=>0,'msg'=>'成功']);
    } else {
      if($img!='') FileEo::RemoveAll($path.$img);
      return self::GetJSON(['code'=>5000,'msg'=>'更新失败!']);
    }
  }

  /* 删除 */
  static function Del() {
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $data = self::JsonName($json, 'data');
    // 验证
    $msg = AdminToken::Verify($token, $_SERVER['REQUEST_URI']);
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($data)) {
      return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    }
    // 数据
    $param = json_decode($data);
    $ids = implode(',',$param);
    // 封面图
    $m = new WebNewsM();
    $m->Columns('id', 'img');
    $m->Where('id in('.$ids.')');
    $imgList = $m->Find();
    // 模型
    $model = new WebNewsM();
    $conn = $model->DBConn();
    try {
      $conn->beginTransaction();
      // 信息
      $m1 = new WebNewsM();
      $m1->Where('id in('.$ids.')');
      list($sql, $args) = $m1->DeleteSQL();
      $model->Exec($conn, $sql, $args);
      // 内容
      $m2 = new WebNewsHtml();
      $m2->Where('nid in('.$ids.')');
      list($sql, $args) = $m2->DeleteSQL();
      $model->Exec($conn, $sql, $args);
      // 提交
      $conn->commit();
      // 清理图片
      foreach($imgList as $v){
        FileEo::RemoveAll($v['img']);
        FileEo::RemoveAll(self::$ImgDir.$v['id'].'/');
      }
      $res = ['code'=>0,'msg'=>'成功'];
    } catch (\Exception $e) {
      $conn->rollBack();
      $res = ['code'=>5000,'msg'=>'删除失败!'];
    }
    // 返回
    return self::GetJSON($res);
  }

  /* 状态 */
  static function State(){
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $id = self::JsonName($json, 'id');
    $state = self::JsonName($json, 'state');
    // 验证
    $msg = AdminToken::Verify($token, $_SERVER['REQUEST_URI']);
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($id)) {
      return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    }
    // 模型
    $m = new WebNewsM();
    $m->Set(['state'=>$state?'1':'0', 'utime'=>time()]);
    $m->Where('id=?', $id);
    if($m->Update()) {
      return self::GetJSON(['code'=>0,'msg'=>'成功']);
    } else {
      return self::GetJSON(['code'=>5000,'msg'=>'更新失败!']);
    }
  }

  /* 分类-获取 */
  static function GetClass(){
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    // 验证
    $msg = AdminToken::Verify($token, '');
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    // 查询
    $m = new WebNewsClass();
    $m->Columns('id', 'name');
    $m->Where('state=?', '1');
    $m->Order('sort DESC');
    $list = $m->Find();
    // 数据
    $data = [];
    foreach ($list as $v) {
      $data[] = ['label'=>$v['name'], 'value'=>$v['id']];
    }
    // 返回
    return self::GetJSON(['code'=>0,'msg'=>'成功','data'=>$data]);
  }

  /* 内容-获取 */
  static function GetContent(){
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $nid = self::JsonName($json, 'id');
    // 验证
    $msg = AdminToken::Verify($token, '');
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    // 查询
    $m = new WebNewsHtml();
    $m->Columns('content');
    $m->Where('nid=?', $nid);
    $data = $m->FindFirst();
    // 返回
    return self::GetJSON(['code'=>0,'msg'=>'成功','content'=>$data['content']]);
  }

  /* 内容-修改 */
  static function Content(){
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $data = self::JsonName($json, 'data');
    // 验证
    $msg = AdminToken::Verify($token, '');
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($data)) {
      return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    }
    // 数据
    $param = json_decode($data);
    $id = isset($param->id)?trim($param->id):'';
    $content = isset($param->content)?trim($param->content):'';
    // 图片回收
    Upload::HtmlImgClear($content, self::$ImgDir.$id.'/');
    // 模型
    $m = new WebNewsHtml();
    $m->Set(['content'=>$content]);
    $m->Where('nid=?', $id);
    if($m->Update()) {
      return self::GetJSON(['code'=>0,'msg'=>'成功']);
    } else {
      return self::GetJSON(['code'=>5000,'msg'=>'更新失败!']);
    }
  }

  /* 内容-图片 */
  static function UpImg(){
    // 参数
    $json = self::Json();
    $token = self::JsonName($json, 'token');
    $base64 = self::JsonName($json, 'base64');
    $id = self::JsonName($json, 'id');
    // 验证
    $msg = AdminToken::Verify($token, '');
    if($msg != '') return self::GetJSON(['code'=>4001, 'msg'=>$msg]);
    if(empty($base64)) return self::GetJSON(['code'=>4000, 'msg'=>'参数错误!']);
    // 上传
    $path = self::$ImgDir.$id.'/';
    $img = Upload::Base64(['path'=>$path, 'base64'=>$base64]);
    if(empty($img)) return self::GetJSON(['code'=>5000, 'msg'=>'上传失败!']);
    // 返回
    return self::GetJSON(['code'=>0,'msg'=>'成功', 'img'=>Data::Img($path.$img)]);
  }

}